trade-finance-manager-api/src/v1/api.js

```diff
@@ -269,6 +269,33 @@ const submitDeal = async (dealType, dealId, auditDetails) => {
   }
 };
 
+/**
+ * @param {object} params
+ * @param {string} params.dealId - deal cancellation to update
+ * @param {Partial<import('@ukef/dtfs2-common').TfmDealCancellation>} params.dealCancellationUpdate - deal cancellation update to make
+ * @param {import('@ukef/dtfs2-common').AuditDetails} params.auditDetails - user making the request
+ * @returns {Promise<import('mongodb').UpdateResult>} update result object
+ */
+const updateDealCancellation = async ({ dealId, dealCancellationUpdate, auditDetails }) => {
+  const isValidDealId = isValidMongoId(dealId);
```

Comment:
Can we wrap this in `try-catch` block please.

Reply:
The only place where this is called and will ever be called is the controller in tfm-api and that wraps this in a `try/catch` block, so it would be unnecessary repetition to also have it here

Reply:
I left the same comment - would also like a try/catch here please like the majority of other functions in this file; Helpful for debugging even if it's only called in one place

Reply:
This came up a fair amount almost a year ago about the error handling in these files and how it's not very good - we're returning a hand rolled status object instead of letting the axios error propagate and losing what axios gives us for free in the error object. The benefit of that as well is with the axios error it's very obvious where the error came from.
If the answer here is we want a consistent (but genuinely worse imo) pattern in these files that's fine but feel the need to point out the above in these files. (This applies to the other api.js file in tfm-ui too but won't copy paste a comment :D )

Reply:
Makes sense to me @ChristianMcc 👍

Reply:
Yep part of my points were about the custom error handling rather than the rethrowing, but even on the the catch and re-throw we do lose access to the stack trace (without some custom code being written). The stack trace can often be very useful for debugging if you are a few functions deep so where possible I'd still avoid catching and re-throwing unless we really really need some extra context in that particular function/extra behaviour

Reply:
@abhi-markan @ttbarnes I've added a try/catch block here as discussed on Friday 👍